plugin/pulp/plugin/task_system.py

```diff
@@ -0,0 +1,34 @@
+from pulp.platform.models import Task
+from pulp.tasking.system import get_current_task_id
+
+
+class TaskController(object):
```

Comment:
I renamed it to `plugin/pulp/plugin/tasking.py`. Good idea. Fixed in my next push.

This object helps me avoid giving access the `Task` model to plugin writers. I am hoping we don't have to include `Task` in the plugin API. The tasking system relies on that data for correctness so I'm also hoping to not give that access to that model for this reason too.

This object can collect any plugin-appropriate Task related calls can live. For now the only one I think we have is to record non-fatal exceptions as a task runs.

I'm removing the `_and_save_` part and I updated the docstring too. The other name was too verbose.